From 0b6f60f1081f7e3df47f88dec2d5426f98fead29 Mon Sep 17 00:00:00 2001
From: Ben Greear <greearb@candelatech.com>
Date: Tue, 21 Jul 2020 15:09:40 -0700
Subject: [PATCH] wlan-ap-ct: Support compile options needed by LANforge

LANforge needs perl and some additional libraries to have a chance
of working on openwrt.

Signed-off-by: Ben Greear <greearb@candelatech.com>
---
 profiles/wlan-ap-ct.yml | 246 ++++++++++++++++++++++++++++++++++++++++
 1 file changed, 246 insertions(+)
 create mode 100644 profiles/wlan-ap-ct.yml

diff --git a/profiles/wlan-ap-ct.yml b/profiles/wlan-ap-ct.yml
new file mode 100644
index 0000000000..2e4de162d8
--- /dev/null
+++ b/profiles/wlan-ap-ct.yml
@@ -0,0 +1,246 @@
+---
+description: Add the wlan-ap dependencies
+feeds:
+  - name: wlan_ap
+    path: ../../feeds/wlan-ap
+
+packages:
+  - cgi-io
+  - coreutils
+  - coreutils-sleep
+  - jansson
+  - kmod-crypto-crc32c
+  - kmod-crypto-hash
+  - kmod-lib-crc32c
+  - kmod-mpls
+  - kmod-nf-nat6
+  - kmod-openvswitch
+  - libatomic
+  - libcares
+  - libev
+  - libiwinfo-lua
+  - liblua
+  - liblucihttp
+  - liblucihttp-lua
+  - libmosquitto-ssl
+  - libopenssl
+  - libpcap
+  - libprotobuf-c
+  - librt
+  - libstdcpp
+  - libubus-lua
+  - libunbound-light
+  - libuuid
+  - lua
+  - protobuf
+  - protobuf-lite
+  - rpcd
+  - rpcd-mod-file
+  - rpcd-mod-iwinfo
+  - rpcd-mod-luci
+  - rpcd-mod-rrdns
+  - uhttpd
+  - uuidgen
+  - zlib
+  - opensync
+  - wpad-openssl
+  - ip-bridge
+  - pciutils
+  - perl
+  - sqlite3
+  - libpcsclite
+
+diffconfig: |
+  CONFIG_OPENSSL_ENGINE=y
+  CONFIG_OPENSSL_PREFER_CHACHA_OVER_GCM=y
+  CONFIG_OPENSSL_WITH_ASM=y
+  CONFIG_OPENSSL_WITH_CHACHA_POLY1305=y
+  CONFIG_OPENSSL_WITH_CMS=y
+  CONFIG_OPENSSL_WITH_DEPRECATED=y
+  CONFIG_OPENSSL_WITH_ERROR_MESSAGES=y
+  CONFIG_OPENSSL_WITH_PSK=y
+  CONFIG_OPENSSL_WITH_SRP=y
+  CONFIG_OPENSSL_WITH_TLS13=y
+  # CONFIG_PACKAGE_wpad-basic is not set
+  CONFIG_BUSYBOX_CONFIG_DC=y
+  CONFIG_BUSYBOX_CONFIG_DOS2UNIX=y
+  CONFIG_BUSYBOX_CONFIG_FEATURE_CATV=y
+  CONFIG_BUSYBOX_CONFIG_FEATURE_DC_LIBM=y
+  CONFIG_BUSYBOX_CONFIG_FEATURE_TELNET_AUTOLOGIN=y
+  CONFIG_BUSYBOX_CONFIG_FEATURE_TELNET_TTYPE=y
+  CONFIG_BUSYBOX_CONFIG_FEATURE_TELNET_WIDTH=y
+  CONFIG_BUSYBOX_CONFIG_FIRST_SYSTEM_ID=0
+  CONFIG_BUSYBOX_CONFIG_HOSTNAME=y
+  CONFIG_BUSYBOX_CONFIG_IFENSLAVE=y
+  # CONFIG_BUSYBOX_CONFIG_IP is not set
+  CONFIG_BUSYBOX_CONFIG_LAST_ID=0
+  CONFIG_BUSYBOX_CONFIG_LAST_SYSTEM_ID=0
+  CONFIG_BUSYBOX_CONFIG_NSLOOKUP=y
+  CONFIG_BUSYBOX_CONFIG_PKILL=y
+  # CONFIG_BUSYBOX_CONFIG_PS is not set
+  CONFIG_BUSYBOX_CONFIG_PSTREE=y
+  CONFIG_BUSYBOX_CONFIG_STAT=y
+  CONFIG_BUSYBOX_CONFIG_SU=y
+  CONFIG_BUSYBOX_CONFIG_TELNET=y
+  CONFIG_BUSYBOX_CONFIG_UNLINK=y
+  CONFIG_BUSYBOX_CONFIG_USLEEP=y
+  CONFIG_BUSYBOX_CONFIG_WHOAMI=y
+  CONFIG_ETHTOOL_PRETTY_DUMP=y
+  CONFIG_OPENSSL_WITH_DEPRECATED=y
+  CONFIG_OPENSSL_WITH_EC=y
+  CONFIG_OPENSSL_WITH_NPN=y
+  CONFIG_OPENSSL_WITH_PSK=y
+  CONFIG_OPENSSL_WITH_SRP=y
+  # CONFIG_PACKAGE_ath10k-firmware-qca9984-ct is not set
+  CONFIG_PACKAGE_ath10k-firmware-qca9984-ct-htt=y
+  CONFIG_PACKAGE_chrony=y
+  CONFIG_PACKAGE_ethtool=y
+  CONFIG_PACKAGE_hostapd-basic=y
+  CONFIG_PACKAGE_ip-bridge=y
+  CONFIG_PACKAGE_ip-full=y
+  CONFIG_PACKAGE_iperf3=y
+  CONFIG_PACKAGE_isc-dhcp-client-ipv4=y
+  CONFIG_PACKAGE_isc-dhcp-server-ipv4=y
+  # CONFIG_PACKAGE_iw is not set
+  CONFIG_PACKAGE_iw-full=y
+  CONFIG_PACKAGE_kmod-bonding=y
+  CONFIG_PACKAGE_kmod-macvlan=y
+  CONFIG_PACKAGE_kmod-pktgen=y
+  CONFIG_PACKAGE_kmod-rtl8812au-ct=y
+  CONFIG_PACKAGE_kmod-usb-acm=y
+  CONFIG_PACKAGE_kmod-usb-serial=y
+  CONFIG_PACKAGE_kmod-usb-serial-ftdi=y
+  CONFIG_PACKAGE_kmod-usb-storage=y
+  CONFIG_PACKAGE_kmod-veth=y
+  CONFIG_PACKAGE_libbz2=y
+  CONFIG_PACKAGE_libcap=y
+  CONFIG_PACKAGE_libexpat=y
+  CONFIG_PACKAGE_libkmod=y
+  CONFIG_PACKAGE_liblzma=y
+  CONFIG_PACKAGE_libmnl=y
+  CONFIG_PACKAGE_libncurses=y
+  CONFIG_PACKAGE_libnet-1.2.x=y
+  CONFIG_PACKAGE_libnl=y
+  CONFIG_PACKAGE_libnl-core=y
+  CONFIG_PACKAGE_libnl-genl=y
+  CONFIG_PACKAGE_libnl-nf=y
+  CONFIG_PACKAGE_libnl-route=y
+  CONFIG_PACKAGE_libopenssl=y
+  CONFIG_PACKAGE_libpcap=y
+  CONFIG_PACKAGE_libpcsclite=y
+  CONFIG_PACKAGE_libreadline=y
+  CONFIG_PACKAGE_librt=y
+  CONFIG_PACKAGE_libsasl2=y
+  CONFIG_PACKAGE_libsqlite3=y
+  CONFIG_PACKAGE_libstdcpp=y
+  CONFIG_PACKAGE_libusb-1.0=y
+  # CONFIG_PACKAGE_odhcp6c is not set
+  # CONFIG_PACKAGE_odhcpd-ipv6only is not set
+  CONFIG_PACKAGE_pciutils=y
+  CONFIG_PACKAGE_perl=y
+  CONFIG_PACKAGE_perl-encode-locale=y
+  CONFIG_PACKAGE_perl-file-listing=y
+  CONFIG_PACKAGE_perl-html-parser=y
+  CONFIG_PACKAGE_perl-html-tagset=y
+  CONFIG_PACKAGE_perl-http-cookies=y
+  CONFIG_PACKAGE_perl-http-daemon=y
+  CONFIG_PACKAGE_perl-http-date=y
+  CONFIG_PACKAGE_perl-http-message=y
+  CONFIG_PACKAGE_perl-http-negotiate=y
+  CONFIG_PACKAGE_perl-io-html=y
+  CONFIG_PACKAGE_perl-lwp-mediatypes=y
+  CONFIG_PACKAGE_perl-net-http=y
+  CONFIG_PACKAGE_perl-net-telnet=y
+  CONFIG_PACKAGE_perl-sub-uplevel=y
+  CONFIG_PACKAGE_perl-test-warn=y
+  CONFIG_PACKAGE_perl-uri=y
+  CONFIG_PACKAGE_perl-www=y
+  CONFIG_PACKAGE_perl-www-robotrules=y
+  CONFIG_PACKAGE_perl-xml-parser=y
+  CONFIG_PACKAGE_perlbase-anydbm-file=y
+  CONFIG_PACKAGE_perlbase-b=y
+  CONFIG_PACKAGE_perlbase-base=y
+  CONFIG_PACKAGE_perlbase-benchmark=y
+  CONFIG_PACKAGE_perlbase-bigint=y
+  CONFIG_PACKAGE_perlbase-bignum=y
+  CONFIG_PACKAGE_perlbase-bytes=y
+  CONFIG_PACKAGE_perlbase-class=y
+  CONFIG_PACKAGE_perlbase-compress=y
+  CONFIG_PACKAGE_perlbase-config=y
+  CONFIG_PACKAGE_perlbase-cwd=y
+  CONFIG_PACKAGE_perlbase-data=y
+  CONFIG_PACKAGE_perlbase-diagnostics=y
+  CONFIG_PACKAGE_perlbase-digest=y
+  CONFIG_PACKAGE_perlbase-dynaloader=y
+  CONFIG_PACKAGE_perlbase-encode=y
+  CONFIG_PACKAGE_perlbase-env=y
+  CONFIG_PACKAGE_perlbase-errno=y
+  CONFIG_PACKAGE_perlbase-essential=y
+  CONFIG_PACKAGE_perlbase-fcntl=y
+  CONFIG_PACKAGE_perlbase-feature=y
+  CONFIG_PACKAGE_perlbase-file=y
+  CONFIG_PACKAGE_perlbase-filehandle=y
+  CONFIG_PACKAGE_perlbase-getopt=y
+  CONFIG_PACKAGE_perlbase-i18n=y
+  CONFIG_PACKAGE_perlbase-integer=y
+  CONFIG_PACKAGE_perlbase-io=y
+  CONFIG_PACKAGE_perlbase-json-pp=y
+  CONFIG_PACKAGE_perlbase-list=y
+  CONFIG_PACKAGE_perlbase-locale=y
+  CONFIG_PACKAGE_perlbase-math=y
+  CONFIG_PACKAGE_perlbase-meta-notation=y
+  CONFIG_PACKAGE_perlbase-mime=y
+  CONFIG_PACKAGE_perlbase-net=y
+  CONFIG_PACKAGE_perlbase-params=y
+  CONFIG_PACKAGE_perlbase-pod=y
+  CONFIG_PACKAGE_perlbase-posix=y
+  CONFIG_PACKAGE_perlbase-re=y
+  CONFIG_PACKAGE_perlbase-scalar=y
+  CONFIG_PACKAGE_perlbase-selectsaver=y
+  CONFIG_PACKAGE_perlbase-selfloader=y
+  CONFIG_PACKAGE_perlbase-sigtrap=y
+  CONFIG_PACKAGE_perlbase-socket=y
+  CONFIG_PACKAGE_perlbase-sort=y
+  CONFIG_PACKAGE_perlbase-symbol=y
+  CONFIG_PACKAGE_perlbase-sys=y
+  CONFIG_PACKAGE_perlbase-tap=y
+  CONFIG_PACKAGE_perlbase-term=y
+  CONFIG_PACKAGE_perlbase-test=y
+  CONFIG_PACKAGE_perlbase-text=y
+  CONFIG_PACKAGE_perlbase-thread=y
+  CONFIG_PACKAGE_perlbase-threads=y
+  CONFIG_PACKAGE_perlbase-tie=y
+  CONFIG_PACKAGE_perlbase-time=y
+  CONFIG_PACKAGE_perlbase-unicore=y
+  CONFIG_PACKAGE_perlbase-utf8=y
+  CONFIG_PACKAGE_perlbase-xsloader=y
+  CONFIG_PACKAGE_procps-ng=y
+  CONFIG_PACKAGE_procps-ng-free=y
+  CONFIG_PACKAGE_procps-ng-kill=y
+  CONFIG_PACKAGE_procps-ng-pgrep=y
+  CONFIG_PACKAGE_procps-ng-pkill=y
+  CONFIG_PACKAGE_procps-ng-pmap=y
+  CONFIG_PACKAGE_procps-ng-ps=y
+  CONFIG_PACKAGE_procps-ng-pwdx=y
+  CONFIG_PACKAGE_procps-ng-skill=y
+  CONFIG_PACKAGE_procps-ng-slabtop=y
+  CONFIG_PACKAGE_procps-ng-snice=y
+  CONFIG_PACKAGE_procps-ng-tload=y
+  CONFIG_PACKAGE_procps-ng-top=y
+  CONFIG_PACKAGE_procps-ng-uptime=y
+  CONFIG_PACKAGE_procps-ng-vmstat=y
+  CONFIG_PACKAGE_procps-ng-w=y
+  CONFIG_PACKAGE_procps-ng-watch=y
+  CONFIG_PACKAGE_strace=y
+  CONFIG_PACKAGE_tcpdump=y
+  CONFIG_PACKAGE_terminfo=y
+  # CONFIG_PACKAGE_wpad-basic is not set
+  CONFIG_PACKAGE_zlib=y
+  CONFIG_PERL_NOCOMMENT=y
+  CONFIG_PERL_THREADS=y
+  CONFIG_SQLITE_FTS3=y
+  CONFIG_SQLITE_RTREE=y
+  CONFIG_BUSYBOX_CONFIG_ASH_BASH_SOURCE_CURDIR=y
+  CONFIG_BUSYBOX_CONFIG_BASH_IS_ASH=y
+  # CONFIG_BUSYBOX_CONFIG_BASH_IS_NONE is not set
+
-- 
2.20.1

